File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/survey/SurveyQuestionMapper.java
package com.tu.votingapp.utils.mappers.survey;

import com.tu.votingapp.dto.general.survey.SurveyQuestionDTO;
import com.tu.votingapp.entities.surveys.SurveyQuestionsEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = {SurveyOptionMapper.class})
public interface SurveyQuestionMapper {

    SurveyQuestionDTO toDto(SurveyQuestionsEntity entity);

    SurveyQuestionsEntity toEntity(SurveyQuestionDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/SurveyQuestionMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.survey.SurveyOptionDTO
import com.example.android.dto.response.survey.SurveyQuestionDTO
import com.example.android.entity.survey.SurveyQuestionEntity

object SurveyQuestionMapper {
    fun SurveyQuestionDTO.toEntity(surveyId: Long): SurveyQuestionEntity = SurveyQuestionEntity(
        id = this.id,
        questionText = this.questionText,
        surveyId = surveyId
    )

    fun SurveyQuestionEntity.toDto(options: List<SurveyOptionDTO>): SurveyQuestionDTO = SurveyQuestionDTO(
        id = this.id!!,
        questionText = this.questionText,
        options = options
    )
}